src/ipv6/ndpv6.ml

```diff
@@ -777,6 +840,22 @@ module Parser = struct
           {pfx_on_link; pfx_autonomous; pfx_valid_lifetime; pfx_preferred_lifetime; pfx_prefix}
         in
         PREFIX pfx :: parse_options1 opts
+      | 25, 3 ->
+        let rdnss_lifetime =
+          let n = Ipv6_wire.get_opt_rdnss_rdnss_lifetime opt in
+          match n with
+          | 0l -> None
+          | n -> Some (Int64.of_int32 n)
+        in
+        let rdnss_addresses =
```

Comment:
here the decoding happens, and `len - 1` needs to be looked at to figure out how many ip addresses to parse. all of them are IPv6 addresses, that's why I'd use Ipaddr.V6.t.

Reply:
If we remove from the `type opt_rdnss` the `rdnss_addresses`, we could:
```OCaml
let decode_ns off = Ipaddr_cstruct.V6.of_cstruct_exn (Cstruct.shift opt off) in
let rec collect_ns acc = function
   | 0 -> acc
   | n ->
    let ns = decode_ns (Ipv6_wire.sizeof_opt_rdnss + n * 16) in
    collect_ns (ns :: acc) (n - 1)
in
let rdnss_addresses = collect_ns [] (Ipv6_wire.get_opt_rdnss_len opt - 1) in
```